sw/device/tests/aon_timer_wdog_bite_reset_test.c

```diff
@@ -0,0 +1,104 @@
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+// TODO:(lowrisc/opentitan#XXXX) Move these difinitions to somewhere else?
+// Power on reset (POR).
+#define RSTMGR_REASON_POWERON (1 << 0)
+// Low power exit (LOW_POWER_EXIT).
+#define RSTMGR_REASON_LOWPOWER_EXIT (1 << 1)
+//  Non-debug module (NDM).
+#define RSTMGR_REASON_NON_DEBUG_MODULE (1 << 2)
+//  Software issued request (SW).
+#define RSTMGR_REASON_SOFTWARE_REQUEST (1 << 3)
+// Hardware requests (HW_REQ).
+#define RSTMGR_REASON_SYSRST_CTRL (1 << 4)
+#define RSTMGR_REASON_WATCHDOG (1 << 5)
+#define RSTMGR_REASON_POWER_UNSTABLE (1 << 6)
+#define RSTMGR_REASON_ESCALATION (1 << 7)
+
+const test_config_t kTestConfig;
+
+/**
+ * Execute the aon timer wdog bite reset test.
+ */
+static void execute_test(dif_aon_timer_t *aon_timer, dif_pwrmgr_t *pwrmgr,
+                         uint32_t bark_time_us) {
+  uint32_t bark_cycles = (bark_time_us * kClockFreqAonHz / 1000000);
+  uint32_t bite_cycles = bark_cycles * 2;
+  uint32_t bite_time_us = bark_time_us * 2;
+
+  LOG_INFO("Wdog will bark after %u us and bite after %u us", bark_time_us,
+           bite_time_us);
+
+  // Set wdog as a reset source.
+  CHECK_DIF_OK(dif_pwrmgr_set_request_sources(
+      pwrmgr, kDifPwrmgrReqTypeReset, kDifPwrmgrWakeupRequestSourceTwo));
+
+  // Setup the wdog bark and bite timeouts.
+  aon_timer_testutils_watchdog_config(aon_timer, bark_cycles, bite_cycles);
+
+  // Wait bark time and check if the bark interrupt fired.
+  usleep(bark_time_us);
+  bool is_pending = false;
+  CHECK_DIF_OK(dif_aon_timer_irq_is_pending(
+      aon_timer, kDifAonTimerIrqWdogTimerBark, &is_pending));
+  CHECK(is_pending);
+
+  // Wait for the remaining time to the wdog bite.
+  usleep(bite_time_us - bark_time_us);
+  // If we arrive here the test must fail.
+  CHECK(false, "Timeout waiting Wdog bite reset!");
+}
+
+bool test_main(void) {
+  // Initialize pwrmgr.
+  dif_pwrmgr_t pwrmgr;
+  CHECK_DIF_OK(dif_pwrmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR), &pwrmgr));
+
+  // Initialize rstmgr to check the reset reason.
+  dif_rstmgr_t rstmgr;
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+
+  // Initialize aon timer to use the wdog.
+  dif_aon_timer_t aon_timer;
+  CHECK_DIF_OK(dif_aon_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR), &aon_timer));
+
+  // Check if there was a HW reset caused by the wdog bite.
+  dif_rstmgr_reset_info_bitfield_t rst_info;
+  CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &rst_info));
+  CHECK_DIF_OK(dif_rstmgr_reset_info_clear(&rstmgr));
+
+  CHECK(rst_info & (RSTMGR_REASON_WATCHDOG | RSTMGR_REASON_POWERON),
+        "Wrong reset reason");
```

Comment:
At the moment, we don't check that this isn't e.g. `RSTMGR_REASON_POWERON | RSTMGR_REASON_POWER_UNSTABLE`. Should this be something more like `rst_info == RSTMGR_REASON_WATCHDOG || rst_info == RSTMGR_REASON_POWERON` to make sure we've got exactly one bit set, which is one of the bits we're expecting?

(And, if so, we can change the `if` tests below to `==`)